etna/ciim/client.py

```diff
@@ -69,32 +70,42 @@ class Aggregation(str, enum.Enum):
     LEVEL = "level"
     CLOSURE = "closure"
     CATALOGUE_SOURCE = "catalogueSource"
+    HELD_BY = "heldBy"
 
 
-def format_list_param(items: Optional[list]) -> Optional[str]:
-    """Convenience function to transform list to comma-separated string.
-
-    When parsing a request's parameters `requests` uses `urllib.parse.urlencode`
-    with the deseq flag set to true, transforming a list's values into multiple
-    parameters:
-
-    >>> urllib.parse.urlencode({'param': ['item-1', 'item-2']}, doseq=True)
-    'param=item-1&param=item-2'
-
-    https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode
-
-    Kong expects multiple values to be a in a comma-separated string:
-
-    'param=item-1,+item-2'
-
-    This function, when given a list, will return a comma-separated string.
-
-    Additionally, to overcome an issue with the API, any commas within paramter
-    text will be removed.
+def prepare_filter_aggregations(items: Optional[list]) -> Optional[str]:
+    """
+    Filter format in items: 'field:value'
+    Prepares i.e. removes/replaces special chars from a filter fields' value to be passed to the api
+
+    Example:
+    before-prepare: "heldBy:Birmingham: Archives, Heritage and Photography Service"
+    after-prepare:  "heldBy:Birmingham  Archives  Heritage and Photography Service"
+    before-prepare: "heldBy:Staffordshire and Stoke-on-Trent Archive Service: Staffordshire County Record Office"
+    after-prepare:  "heldBy:Staffordshire and Stoke on Trent Archive Service  Staffordshire County Record Office"
+    Special char single quote i.e. ' is not prepared
+    before-prepare: "heldBy:Labour History Archive and Study Centre (People's History Museum/University of Central Lancashire)"
+    after-prepare:  "heldBy:Labour History Archive and Study Centre  People's History Museum University of Central Lancashire "
+    before-prepare: "heldBy:Bedfordshire Archives & Records Service"
+    after-prepare: "heldBy:Bedfordshire Archives   Records Service"
     """
     if not items:
         return None
-    return ", ".join([i.replace(",", "") for i in items])
+
+    regex = r"([/():,&-])"
+    subst = " "
+    field_list_to_prepare = ["heldBy"]
+    filter_prepared_list = []
+
+    for item in items:
+        field, value = item.split(":", 1)
+        if field in field_list_to_prepare:
+            filter_prepared = field + ":" + re.sub(regex, subst, value, 0, re.MULTILINE)
+        else:
+            filter_prepared = field + ":" + value
```

Comment:
I think this is the right choice for the client code... Even though ':' can be problematic elsewhere, that is definitely something that should be handled by the API.